File: Src/Icm.Core.Tests/CompositeWriterTest.vb
Imports System.Text
Imports System.IO
Imports System.IO.StringWriter
Imports Icm.IO



'''<summary>
'''This is a test class for CompositeWriterTest and is intended
'''to contain all CompositeWriterTest Unit Tests
'''</summary>
<TestFixture(), Category("Icm")>
Public Class CompositeWriterTest

    '''<summary>
    '''A test for WriteLine
    '''</summary>
    <Test()>
    Public Sub CompositeWriterTest()
        Dim target As CompositeWriter = New CompositeWriter
        Dim s As String = String.Empty

        s = "hola"
        Dim sw1 As New StringWriter()

        target.Add(sw1)
        target.Write(s)

        Assert.That(sw1.ToString = "hola")

        Dim sw2 As New StringWriter()
        target.Add(sw2)
        target.WriteLine(Date.Now)
        'Debug.WriteLine(sw1)
        Dim aux2 = "hola" & sw2.ToString
        Assert.That(sw1.ToString = aux2)

        Dim sw3 As New StringWriter
        target.Add(sw3)
        target.Write("Prueba")
        Assert.That(sw1.ToString = "hola" & sw2.ToString)
        Assert.That(sw3.ToString = "Prueba")
        target.Close()


    End Sub

End Class

File: Src/Icm.Core.Tests/CompositeWriterTest.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Text;
using System.IO;
using System.IO.StringWriter;
using Icm.IO;

[TestFixture(), Category("Icm")]
public class CompositeWriterTest
{

	///<summary>
	///A test for WriteLine
	///</summary>
	[Test()]
	public void Write_Test()
	{
		CompositeWriter target = new CompositeWriter();
		dynamic s1 = "hola";
		dynamic s2 = "quetal";
		dynamic s3 = "adios";
		StringWriter sw1 = new StringWriter();
		StringWriter sw2 = new StringWriter();
		StringWriter sw3 = new StringWriter();

		target.Add(sw1);
		target.Write(s1);

		Assert.That(sw1.ToString == "hola");

		target.Add(sw2);
		target.Write(s2);

		Assert.That(sw1.ToString == "holaquetal");
		Assert.That(sw2.ToString == "quetal");

		target.Add(sw3);
		target.Write(s3);
		Assert.That(sw1.ToString == "holaquetaladios");
		Assert.That(sw2.ToString == "quetaladios");
		Assert.That(sw3.ToString == "adios");
		target.Close();
	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
